### ISSUE TITLE
ERROR: the following samples are listed in the population file but not in the VCF

### ISSUE THREAD

**Issue by tshalev** [ksamuk/pixy#17]
Hi Kieran!

I am trying to use pixy with the following command:

`pixy --stats pi dxy fst --vcf diverse_pop.scaffolds_withgenes_final_annotation.recode.vcf.gz --zarr_path zarr --populations parents_popmap_pixy.txt --bypass_filtration yes --outfile_prefix output/pixy_out` 

However, I keep getting the following error:

`Traceback (most recent call last):
  File "/mnt/c/Users/tal_shalev/software/anaconda3/lib/python3.8/site-packages/pixy/__main__.py", line 200, in main
    samples_callset_index = [samples_list.index(s) for s in poppanel['ID']]
  File "/mnt/c/Users/tal_shalev/software/anaconda3/lib/python3.8/site-packages/pixy/__main__.py", line 200, in <listcomp>
    samples_callset_index = [samples_list.index(s) for s in poppanel['ID']]
ValueError: 367 is not in list

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/c/Users/tal_shalev/software/anaconda3/bin/pixy", line 11, in <module>
    sys.exit(main())
  File "/mnt/c/Users/tal_shalev/software/anaconda3/lib/python3.8/site-packages/pixy/__main__.py", line 202, in main
    raise Exception('[pixy] ERROR: the following samples are listed in the population file but not in the VCF: ', missing) from e
Exception: ('[pixy] ERROR: the following samples are listed in the population file but not in the VCF: ', [1, 2, 5, 10, 527, 528, 15, 19, 531, 20, 40, 44, 47, 49, 566, 1081, 58, 85, 87, 88, 94, 95, 97, 113, 116, 120, 121, 126, 128, 133, 648, 137, 138, 144, 148, 149, 150, 662, 156, 158, 164, 165, 171, 173, 175, 176, 181, 191, 202, 203, 204, 214, 220, 221, 230, 231, 232, 235, 241, 243, 253, 261, 263, 264, 266, 779, 784, 281, 285, 301, 306, 322, 325, 329, 331, 334, 336, 340, 343, 344, 347, 351, 366, 367, 368, 378, 381, 386, 392, 394, 402, 404, 411, 415, 421, 426, 430, 435, 438, 444, 446, 454, 461, 465, 466, 490, 493, 497, 501, 503, 505, 508])`

I'm a little confused as all these samples are present in the VCF. I don't have any problem identifying these sample names in the VCF using any other approach, e.g. VCFtools or BCFtools. Have you encountered this before?

Thanks,
Tal

**Comment by ksamuk** [comment]
Hi Tal, that's odd! Can you post your population file? How specifically are the samples named in the VCF?

**Comment by tshalev** [comment]
This is my population file:

`367	Vancouver_Island_Centre
19	Vancouver_Island_North
454	Vancouver_Island_Centre
171	Sunshine_Coast
784	Vancouver_Island_Centre
149	Sunshine_Coast
394	Sunshine_Coast
165	Sunshine_Coast
344	Vancouver_Island_North
446	Vancouver_Island_North
2	Vancouver_Island_South
241	Central_Coast
404	Vancouver_Island_North
87	Haida_Gwaii
381	Vancouver_Island_Centre
148	Sunshine_Coast
648	Oregon
128	Haida_Gwaii
88	Haida_Gwaii
203	Sunshine_Coast
531	Interior_BC
501	Vancouver_Island_Centre
347	Vancouver_Island_North
116	Haida_Gwaii
235	Central_Coast
306	Vancouver_Island_North
150	Sunshine_Coast
49	Vancouver_Island_North
466	Vancouver_Island_North
175	Vancouver_Island_South
421	Central_Coast_Islands
133	Haida_Gwaii
368	Vancouver_Island_Centre
176	Vancouver_Island_South
202	Sunshine_Coast
113	Haida_Gwaii
121	Haida_Gwaii
438	Vancouver_Island_South
173	Vancouver_Island_South
392	Sunshine_Coast
120	Haida_Gwaii
58	Vancouver_Island_North
126	Haida_Gwaii
44	Vancouver_Island_North
329	Vancouver_Island_North
232	Central_Coast
164	Sunshine_Coast
334	Vancouver_Island_North
158	Vancouver_Island_Centre
1	Vancouver_Island_North
264	Sunshine_Coast
402	Vancouver_Island_North
156	Vancouver_Island_Centre
366	Vancouver_Island_Centre
266	Haida_Gwaii
301	Vancouver_Island_North
221	Vancouver_Island_North
340	Vancouver_Island_North
461	Vancouver_Island_North
527	Interior_BC
5	Vancouver_Island_North
281	Haida_Gwaii
779	Vancouver_Island_Centre
508	Central_Coast_Islands
490	Vancouver_Island_Centre
263	Sunshine_Coast
415	Central_Coast
378	Vancouver_Island_Centre
40	Vancouver_Island_North
47	Vancouver_Island_North
243	Central_Coast
97	Haida_Gwaii
465	Vancouver_Island_North
430	Vancouver_Island_South
144	Vancouver_Island_North
503	Vancouver_Island_Centre
493	Vancouver_Island_Centre
285	Vancouver_Island_North
497	Vancouver_Island_South
204	Sunshine_Coast
331	Vancouver_Island_North
94	Haida_Gwaii
336	Vancouver_Island_North
351	Vancouver_Island_South
138	Haida_Gwaii
411	Lower_Mainland
505	Central_Coast_Islands
566	California
10	Vancouver_Island_North
426	Vancouver_Island_South
181	Vancouver_Island_South
528	Interior_BC
325	Vancouver_Island_North
85	Sunshine_Coast
191	Sunshine_Coast
662	Oregon
444	Vancouver_Island_North
261	Central_Coast
230	Central_Coast
220	Vancouver_Island_North
214	Lower_Mainland
322	Vancouver_Island_North
1081	Oregon
95	Haida_Gwaii
20	Vancouver_Island_North
137	Haida_Gwaii
343	Vancouver_Island_North
231	Central_Coast
435	Vancouver_Island_South
15	Vancouver_Island_North
253	Central_Coast
386	Vancouver_Island_Centre`

In the VCF, the header looks like this:

`#CHROM | POS | ID | REF | ALT | QUAL | FILTER | INFO | FORMAT | 367 | 19 | 454 | 171 | 784 | 149 | 394 | 165 | 344 | 446 | 2 | 241 | 404 | 87 | 381 | 148 | 648 | 128 | 88 | 203 | 531 | 501 | 347 | 116 | 235 | 306 | 150 | 49 | 466 | 175 | 421 | 133 | 368 | 176 | 202 | 113 | 121 | 438 | 173 | 392 | 120 | 58 | 126 | 44 | 329 | 232 | 164 | 334 | 158 | 1 | 264 | 402 | 156 | 366 | 266 | 301 | 221 | 340 | 461 | 527 | 5 | 281 | 779 | 508 | 490 | 263 | 415 | 378 | 40 | 47 | 243 | 97 | 465 | 430 | 144 | 503 | 493 | 285 | 497 | 204 | 331 | 94 | 336 | 351 | 138 | 411 | 505 | 566 | 10 | 426 | 181 | 528 | 325 | 85 | 191 | 662 | 444 | 261 | 230 | 220 | 214 | 322 | 1081 | 95 | 20 | 137 | 343 | 231 | 435 | 15 | 253 | 386`

So, looks like identical names and identical order.

**Comment by ksamuk** [comment]
Hi Tal, thanks for posting these details. I've tracked down the issue -- basically pixy in its current form doesn't support sample IDs that are not strings (e.g. integers such as you have). This was news to me, so thanks for pointing out this bug! Anyway, this is now addressed the next update (coming in a couple of weeks). The quick fix on your end is to append some non-numeric characters (e.g. "sample1" instead of "1") to your sample IDS in your VCF and in your samples file.
